linera-core/src/updater.rs

```diff
@@ -391,10 +392,12 @@ where
                 height, chain_id, ..
             } => (*height, *chain_id),
         };
-        // Update the validator with missing information, if needed.
-        let delivery = CrossChainMessageDelivery::NonBlocking;
-        self.send_chain_information(chain_id, target_block_height, delivery)
-            .await?;
+        if !optimistic_client {
+            // Update the validator with missing information, if needed.
+            let delivery = CrossChainMessageDelivery::NonBlocking;
+            self.send_chain_information(chain_id, target_block_height, delivery)
+                .await?;
```

Comment:
Aren't we calling `send_chain_update` also to update the validators about a new certificate that we just created? Because then the validators definitely won't have it.